scripts/source-maintenance.sh

```diff
@@ -317,7 +359,24 @@ git grep "ifn?def .*_SQUID_" |
 #
 # Scan for file-specific actions
 #
-for FILENAME in `git ls-files`; do
+ForkPoint=""
+FilesToOperateOn=""
+if test "x$OnlyChangedSince" = "x" ; then
+    FilesToOperateOn=`git ls-files`
+elif test "x$OnlyChangedSince" = "xfork" ; then
+    ForkPoint=`git merge-base --fork-point upstream/master`
+    if test "x$ForkPoint" = "x" ; then
+        echo "Could not identify fork point - sometimes it happens"
+        echo "Please specify commit-id explicitly"
+        exit 1
+    fi
+    OnlyChangedSince="$ForkPoint"
+fi
+if test "x$FilesToOperateOn" = "x" && test "x$OnlyChangedSince" != "x" ; then
+    FilesToOperateOn=`git diff --name-only $OnlyChangedSince`
+fi
```

Comment:
Please simplify by handling FilesToOperateOn in one place, after the xfork check:

```suggestion
if test "x$OnlyChangedSince" = "xfork" ; then
    ForkPoint=`git merge-base --fork-point upstream/master`
    if test "x$ForkPoint" = "x" ; then
        echo "Could not identify fork point - sometimes it happens"
        echo "Please specify commit-id explicitly"
        exit 1
    fi
    OnlyChangedSince="$ForkPoint"
fi

FilesToOperateOn=""
if test "x$OnlyChangedSince" != "x" ; then
    FilesToOperateOn=`git diff --name-only $OnlyChangedSince`
else
    FilesToOperateOn=`git ls-files`
fi
```